module/zfs/vdev_label.c

```diff
@@ -1161,6 +1230,14 @@ vdev_uberblock_sync(zio_t *zio, uberblock_t *ub, vdev_t *vd, int flags)
 	if (!vdev_writeable(vd))
 		return;
 
+	/* If the vdev was expanded, need to copy uberblock rings. */
+	if (vd->vdev_ops->vdev_op_leaf &&
+	    (vd->vdev_state == VDEV_STATE_HEALTHY) &&
+	    (vd->vdev_copy_uberblocks == B_TRUE)) {
+		if (vdev_copy_uberblocks(vd) == 0)
+			vd->vdev_copy_uberblocks = B_FALSE;
```

Comment:
I think best effort is what we want here.  We shouldn't retry this operation on every `vdev_uberblock_sync()` until it succeeds.  The pool will naturally gravitate towards having all these uberblocks updated regardless.  That would also allow us to make ` vdev_copy_uberblocks()` return a void type and remove that slightly complicated `error / rc` tracking.